### ISSUE TITLE
Write up docs on component integration tests combined with closure actions

### ISSUE THREAD

**Issue by acorncom** [emberjs/guides#1223]
Right now, we don't have anything documented about how to pass in closure actions to a component in an integration test.

**Comment by toddjordan** [comment]
I'm having an internal debate on whether you would want to pass a closure action to your component under test, as opposed to a regular function.

For example:
this:
```
  this.set('onFileUpload', function() {});
  this.render(hbs`{{application-attachments onFileDelete=(action onFileUpload)}}`);
```
is the same to your component under test as this:
```
  this.set('onFileUpload', function() {});
  this.render(hbs`{{application-attachments onFileDelete=onFileUpload}}`);
```
If I understand correctly, the component gets a function either way

by using the action helper you are not testing anything we don't already know works.  I'd be interested in other's thoughts, or corrections to my understanding.

**Comment by jeradg** [comment]
@toddjordan: On the other hand, there's the view that tests also serve as de facto documentation of how to use code. So if a given component is likely to receive an action that's declared in an `actions` hash, then it could be useful to write the test to reflect that.

```
this.set('actions', { onFileUpload: function() {} });
this.render(hbs`{{application-attachments onFileDelete=(action 'onFileUpload')}}`);
```

**Comment by rwjblue** [comment]
FWIW, I generally do this:

```
  this.on('onFileUpload', function() {});
  this.render(hbs`{{application-attachments onFileUpload=(action 'onFileUpload')}}`);
```

**Comment by toddjordan** [comment]
wow.  4 ways to skin this cat :-P  They're all just giving the component a function right?

**Comment by rwjblue** [comment]
`this.on` puts it in `actions` hash

**Comment by toddjordan** [comment]
yeah, but in the actions hash of your test, not the component under test, right?

**Comment by rwjblue** [comment]
confirm

**Comment by sukima** [comment]
Isn't this issue resolved by commit [9a33b31e](https://github.com/emberjs/guides/commit/9a33b31e66b85d78c29a121c3ac91cddefe44790)? Or is this specifically about converting `this.set('actionName', …)` to `this.on('actionName', …)`?
